schema/types/labels.go

```diff
@@ -2,9 +2,16 @@ package types
 
 import (
 	"encoding/json"
-	"errors"
+	"fmt"
+	"sort"
 )
 
+var ValidOSArch = map[string][]string{
+	"linux":   []string{"amd64", "i386"},
+	"freebsd": []string{"amd64", "i386", "arm"},
+	"darwin":  []string{"x86_64", "i386"},
```

Comment:
@jonboulle I see. This is a little unfortunate, since this is just being different to be different, rather than providing a benefit over GOARCH/GOOS.